controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -2565,13 +2610,43 @@ int bml_internal::get_restricted_channels(uint8_t *restricted_channels, const st
     return (iRet);
 }
 
-int bml_internal::wfca_controller(const char *cmd, char *ret_buf, int ret_buf_size)
+int bml_internal::wfa_ca_controller(BML_CTX ctx, const char *command, int command_len,
+                                    const char **first_reply, BML_WFA_CA_CB second_reply_cb)
 {
-    // TODO: implement function
-    return BML_RET_OK;
+    //CMDU message
+    auto request =
+        message_com::create_vs_message<beerocks_message::cACTION_BML_WFA_CA_CONTROLLER_REQUEST>(
+            cmdu_tx);
+
+    if (!request) {
+        LOG(ERROR) << "Failed building cACTION_BML_WFA_CA_CONTROLLER_REQUEST message!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    request->alloc_command(command_len);
```

Comment:
Same here - after you change to char in the yaml:
```
request->set_command(std::string(command), command_len);
```
No need for all of the fancy tuples and buffers, all taken care inside the TLVF.